mapstruct-kotlin/README.md

```diff
@@ -0,0 +1,103 @@
+# MapStruct with Kotlin
+
+Example usage of Mapstruct and Kotlin with JDK-8 Bytecode. This is achieved by using the [Kotlin Annotation Processing Tool (KAPT)](https://kotlinlang.org/docs/reference/kapt.html).
+
+## Code
+
+In this example we want to map between a Person (Model) and a PersonDto (DTO).
+
+```kotlin
+data class Person(var firstName: String?, var lastName: String?, var phoneNumber: String?, var birthdate: LocalDate?) {
+    constructor() : this(null, null, null, null)
+}
+```
+
+```kotlin
+data class PersonDto(var firstName: String?, var lastName: String?, var phone: String?, var birthdate: LocalDate?) {
+    // Necessary for MapStruct
+    constructor() : this(null, null, null, null)
+}
+```
+
+The MapStruct converter:
+
+```kotlin
+@Mapper
+interface PersonConverter {
+
+    @Mapping(source = "phoneNumber", target = "phone")
+    fun convertToDto(person: Person) : PersonDto
+
+    @InheritInverseConfiguration
+    fun convertToModel(personDto: PersonDto) : Person
+
+}
+```
+
+Usage:
+
+```kotlin
+val converter = PersonConverterImpl()
```

Comment:
It is possible, but the mapstruct-jdk8 dependency must have compile-scope, see below